src/ogl/opengl_wrapper.cpp

```diff
@@ -456,12 +456,60 @@ void render_character(sys::state const& state, char codepoint, color_modificatio
 	}
 }
 
-void internal_text_render(sys::state const& state, char const* codepoints, uint32_t count, float x, float baseline_y, float size,
+GLuint get_flag_texture_handle_from_tag(sys::state& state, char tag[3]) {
+	dcon::national_identity_id ident{};
+	state.world.for_each_national_identity([&](dcon::national_identity_id id) {
+		auto curr = nations::int_to_tag(state.world.national_identity_get_identifying_int(id));
+		if(curr[0] == tag[0] && curr[1] == tag[1] && curr[2] == tag[2]) {
+			ident = id;
+		}
+	});
+	if(!bool(ident))
+		ident = state.world.nation_get_identity_from_identity_holder(state.national_definitions.rebel_id);
+	auto fat_id = dcon::fatten(state.world, ident);
+	auto nation = fat_id.get_nation_from_identity_holder();
+	culture::flag_type flag_type = culture::flag_type{};
+	if(bool(nation.id) && nation.get_owned_province_count() != 0) {
+		flag_type = culture::get_current_flag_type(state, nation.id);
+	} else {
+		flag_type = culture::get_current_flag_type(state, ident);
+	}
+	return ogl::get_flag_handle(state, ident, flag_type);
+}
+
+void internal_text_render(sys::state& state, char const* codepoints, uint32_t count, float x, float baseline_y, float size,
 		text::font& f, GLuint const* subroutines, GLuint const* icon_subroutines) {
 	for(uint32_t i = 0; i < count; ++i) {
 		if(text::win1250toUTF16(codepoints[i]) != ' ') {
 			// f.make_glyph(codepoints[i]);
-			if(text::win1250toUTF16(codepoints[i]) != u'\u0001' && text::win1250toUTF16(codepoints[i]) != u'\u0002') {
+			if(text::win1250toUTF16(codepoints[i]) != u'\u0040') {
+				char tag[3] = { 0, 0, 0 };
+				i++;
+				if(i + 1 < count) {
+					tag[0] = char(codepoints[i++]);
+					if(i + 1 < count) {
+						tag[1] = char(codepoints[i++]);
+						if(i + 1 < count) {
+							tag[2] = char(codepoints[i++]);
+						}
+					}
+				}
+				i--;
+				GLuint flag_texture_handle = get_flag_texture_handle_from_tag(state, tag);
```

Comment:
You probably want to do the following: are there more than 3 characters after the `@` and are those 3 a valid tag? If so, render the flag. Otherwise, render the text as text. (and then mirror this into the metrics) This allows misuses of the `@` to show up in text so that they are findable